Kernel/Devices/FUSEDevice.cpp

```diff
@@ -164,40 +168,71 @@ ErrorOr<NonnullOwnPtr<KBuffer>> FUSEDevice::send_request_and_wait_for_a_reply(Op
 
     u64 unique = bit_cast<fuse_in_header*>(bytes.data())->unique;
 
-    while (true) {
-        auto error_or_buffer = m_instances.with([&](auto& instances) -> ErrorOr<NonnullOwnPtr<KBuffer>> {
+    ErrorOr<KBuffer*> reply_or_error = nullptr;
+    auto receive_reply = [&](auto& instances) -> bool {
+        auto instance_iterator = instances.active_instances.find(&description);
+        VERIFY(instance_iterator != instances.active_instances.end());
+        auto& requests_for_instance = (*instance_iterator).value;
+
+        for (size_t i = 0; i < requests_for_instance.size(); ++i) {
+            if (bit_cast<fuse_in_header*>(requests_for_instance[i].pending_request->data())->unique != unique)
+                continue;
+            if (requests_for_instance[i].timed_out) {
+                requests_for_instance.remove(i);
+                reply_or_error = ETIMEDOUT;
+                return true;
+            }
+            if (!requests_for_instance[i].response_ready)
+                continue;
+            auto buffer_or_error = KBuffer::try_create_with_bytes("FUSEDevice: Response"sv, requests_for_instance[i].response->bytes());
+            if (buffer_or_error.is_error())
+                reply_or_error = buffer_or_error.release_error();
+            else
+                reply_or_error = buffer_or_error.release_value().leak_ptr();
+            requests_for_instance.remove(i);
+            return true;
+        }
+
+        return false;
+    };
+
+    auto set_timeout = [&]() {
```

Comment:
nit: `set_timeout` is a weird name for this lambda. Maybe `set_timed_out` instead?